api/database_engine.go

```diff
@@ -90,6 +94,20 @@ func (e *EverestServer) UpgradeDatabaseEngineOperator(ctx echo.Context, namespac
 		return err
 	}
 
+	// Validate preflight checks.
+	preflight, err := e.getOperatorUpgradePreflight(ctx.Request().Context(), req.TargetVersion, name, namespace)
```

Comment:
Is the engine/namespace locked at this point? I could not find a good reference for it.

Reply:
Its not. The lock is put in place only once a DBEngine is upgrading (visible in its status). See reference: https://github.com/percona/everest/pull/201/files#diff-bd0949bd5a83da0c2752e1466c3562542455b2e260def83d578158465b2e59e6R64

Again, there is no explicit locking mechanism for this. Its just an API middleware that checks if there's an upgrade happening in a namespace and blocks requests accordingly..

Reply:
Based on our Slack chat, we will redo the locking mechanism. This is implemented in: https://github.com/percona/everest/pull/201